src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/UploadDocumentContestedAboutToSubmitHandler.java

```diff
@@ -0,0 +1,90 @@
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.UploadGeneralDocument;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.UploadGeneralDocumentCollection;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.NewUploadedDocumentsService;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.documentcatergory.UploadGeneralDocumentsCategoriser;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.documentchecker.DocumentCheckerService;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.nullsLast;
+import static java.util.Optional.ofNullable;
+
+@Slf4j
+@Service
+public class UploadDocumentContestedAboutToSubmitHandler extends FinremCallbackHandler {
+
+    private final DocumentCheckerService documentCheckerService;
+    private final NewUploadedDocumentsService newUploadedDocumentsService;
+    private final UploadGeneralDocumentsCategoriser uploadGeneralDocumentsCategoriser;
+
+    public UploadDocumentContestedAboutToSubmitHandler(FinremCaseDetailsMapper mapper,
+                                                       DocumentCheckerService documentCheckerService,
+                                                       NewUploadedDocumentsService newUploadedDocumentsService,
+                                                       UploadGeneralDocumentsCategoriser uploadGeneralDocumentsCategoriser) {
+        super(mapper);
+        this.documentCheckerService = documentCheckerService;
+        this.newUploadedDocumentsService = newUploadedDocumentsService;
+        this.uploadGeneralDocumentsCategoriser = uploadGeneralDocumentsCategoriser;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_SUBMIT.equals(callbackType)
+            && CaseType.CONTESTED.equals(caseType)
+            && EventType.UPLOAD_DOCUMENT_CONTESTED.equals(eventType);
+    }
+
+    @Override
+    @SuppressWarnings("squid:CallToDeprecatedMethod")
+    public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(FinremCallbackRequest callbackRequest,
+                                                                              String userAuthorisation) {
+        FinremCaseDetails beforeFinremCaseDetails = callbackRequest.getCaseDetailsBefore();
+        FinremCaseDetails finremCaseDetails = callbackRequest.getCaseDetails();
+        FinremCaseData caseData = finremCaseDetails.getData();
+        FinremCaseData caseDataBefore = callbackRequest.getCaseDetailsBefore().getData();
+
+        final Set<String> warnings = newUploadedDocumentsService.getNewUploadDocuments(caseData, caseDataBefore,
```

Comment:
I think it would be sensible to separate out the code to get newly uploaded documents from that which checks for warnings and to wrap the warnings check in an exception block so that any unexpected exceptions don't cause the upload event to fail

Reply:
`    public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(FinremCallbackRequest callbackRequest,
                                                                              String userAuthorisation) {
        FinremCaseDetails beforeFinremCaseDetails = callbackRequest.getCaseDetailsBefore();
        FinremCaseDetails finremCaseDetails = callbackRequest.getCaseDetails();
        FinremCaseData caseData = finremCaseDetails.getData();
        FinremCaseData caseDataBefore = callbackRequest.getCaseDetailsBefore().getData();
        final Long caseId = finremCaseDetails.getId();

        List<UploadGeneralDocumentCollection> newDocuments = newUploadedDocumentsService.getNewUploadDocuments(caseData, caseDataBefore,
            FinremCaseData::getUploadGeneralDocuments);

        Set<String> allWarnings = new HashSet<>();
        for (UploadGeneralDocumentCollection doc : newDocuments) {
            if (doc.getValue() != null && doc.getValue().getDocumentLink() != null) {
                try {
                    List<String> warnings = documentCheckerService.getWarnings(doc.getValue().getDocumentLink(), beforeFinremCaseDetails, finremCaseDetails,
                        userAuthorisation);
                    if (!ObjectUtils.isEmpty(warnings)) {
                        allWarnings.addAll(warnings);
                    }
                } catch (Exception t) {
                    log.error(format("%s - Exception was caught when checking the warnings", caseId), t);
                }
            } else {
                throw new UnsupportedOperationException(format("%s - unexpected null in value or documentLink", caseId));
            }
        }

        if (!allWarnings.isEmpty()) {
            log.info(format("%s - Number of warnings encountered when uploading general document %s",
                caseId, allWarnings.size()));
        }

        // Do sorting
        List<UploadGeneralDocumentCollection> uploadedDocuments = new ArrayList<>(ofNullable(caseData.getUploadGeneralDocuments())
            .orElse(List.of()));
        uploadedDocuments.sort(comparing(UploadGeneralDocumentCollection::getValue,
            comparing(UploadGeneralDocument::getGeneralDocumentUploadDateTime, nullsLast(Comparator.reverseOrder()))));
        caseData.setUploadGeneralDocuments(uploadedDocuments);

        // Execute the categoriser for CFV
        uploadGeneralDocumentsCategoriser.categorise(caseData);

        return GenericAboutToStartOrSubmitCallbackResponse.<FinremCaseData>builder()
            .warnings(allWarnings.stream().sorted().toList())
            .data(caseData).build();
    }`

Reply:
@jthmcts can you please review it before I implement the logic in the consented flow and write unit tests?

Reply:
Improved the logic -  https://github.com/hmcts/finrem-case-orchestration-service/pull/1788/commits/b20fc27a08f4b1a4ac2bab72f8afcf9a9107e90e